Build fails with  'UPDATE_CHECK' disabled.

##### Brief Description

OS: Gentoo Linux
qTox version: `git master`
Commit hash:  69a4f6dca26cece7d738f91acd00af434a1ef636
toxcore:  0.2.20
Qt: 6.8.0

Reproducible: Always

##### Steps to reproduce

1. Configure and build qTox with `-DUPDATE_CHECK=off

##### Observed Behavior

```
FAILED: CMakeFiles/qtox_static.dir/src/widget/widget.cpp.o
/usr/bin/x86_64-pc-linux-gnu-g++ -DCMAKE_BUILD -DGIT_DESCRIBE=\"nightly-9-g69a4f6dc\" -DGIT_DESCRIBE_EXACT=\"Nightly\" -DGIT_VERSION=\"69a4f6dca26cece7d738f91acd00af434a1ef636\" -DLOG_TO_FILE=1 -DQTOX_PLATFORM_EXT -DQT_CORE_LIB -DQT_GUI_LIB -DQT_MESSAGELOGCONTEXT=1 -DQT_NETWORK_LIB -DQT_NO_CAST_FROM_BYTEARRAY -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_OPENGL_LIB -DQT_RESTRICTED_CAST_FROM_ASCII -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -DSPELL_CHECKING -I/mnt/lprup/portage-build/portage/net-im/qtox-9999/work/qtox-9999_build/qtox_static_autogen/include -I/mnt/lprup/portage-build/portage/net-im/qtox-9999/work/qtox-9999_build -I/mnt/lprup/portage-build/portage/net-im/qtox-9999/work/qtox-9999 -I/usr/include/libexif -I/usr/include/sqlcipher -I/usr/include/opus -I/usr/include/AL -I/mnt/lprup/portage-build/portage/net-im/qtox-9999/work/qtox-9999/util/include -I/mnt/lprup/portage-build/portage/net-im/qtox-9999/work/qtox-9999/audio/include -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/lib64/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtGui -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtOpenGL -isystem /usr/include/qt6/QtSvg -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtXml -isystem /usr/include/KF6/SonnetUi -isystem /usr/include/KF6/Sonnet  -march=znver4 -O2 -fomit-frame-pointer -pipe -mindirect-branch=thunk -g -std=c++20 -fno-exceptions -fno-rtti -fstack-protector-all -fPIC -fno-common -fstrict-overflow -ftrapv -pedantic-errors -Wall -Wcast-align -Wdouble-promotion -Wextra -Wformat=2 -Wmissing-declarations -Wnon-virtual-dtor -Wnull-dereference -Wold-style-cast -Woverloaded-virtual -Wshadow -Wsign-compare -Wundef -Wduplicated-cond -Wlogical-op -Wdate-time -Wstack-protector -MD -MT CMakeFiles/qtox_static.dir/src/widget/widget.cpp.o -MF CMakeFiles/qtox_static.dir/src/widget/widget.cpp.o.d -o CMakeFiles/qtox_static.dir/src/widget/widget.cpp.o -c /mnt/lprup/portage-build/portage/net-im/qtox-9999/work/qtox-9999/src/widget/widget.cpp
/mnt/lprup/portage-build/portage/net-im/qtox-9999/work/qtox-9999/src/widget/widget.cpp:290:5: warning: "UPDATE_CHECK_ENABLED" is not defined, evaluates to 0 [-Wundef]
  290 | #if UPDATE_CHECK_ENABLED
      |     ^~~~~~~~~~~~~~~~~~~~
/mnt/lprup/portage-build/portage/net-im/qtox-9999/work/qtox-9999/src/widget/widget.cpp:296:5: warning: "UPDATE_CHECK_ENABLED" is not defined, evaluates to 0 [-Wundef]
  296 | #if UPDATE_CHECK_ENABLED
      |     ^~~~~~~~~~~~~~~~~~~~
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/14/include/g++-v14/memory:78,
                 from /usr/include/qt6/QtCore/qcontainertools_impl.h:21,
                 from /usr/include/qt6/QtCore/qarraydataops.h:9,
                 from /usr/include/qt6/QtCore/qarraydatapointer.h:7,
                 from /usr/include/qt6/QtCore/qbytearray.h:11,
                 from /usr/include/qt6/QtCore/qmetatype.h:11,
                 from /usr/include/qt6/QtCore/qvariant.h:10,
                 from /usr/include/qt6/QtCore/QVariant:1,
                 from /mnt/lprup/portage-build/portage/net-im/qtox-9999/work/qtox-9999_build/ui_mainwindow.h:12,
                 from /mnt/lprup/portage-build/portage/net-im/qtox-9999/work/qtox-9999/src/widget/widget.h:8,
                 from /mnt/lprup/portage-build/portage/net-im/qtox-9999/work/qtox-9999/src/widget/widget.cpp:6:
/usr/lib/gcc/x86_64-pc-linux-gnu/14/include/g++-v14/bits/unique_ptr.h: In instantiation of ‘void std::default_delete<_Tp>::operator()(_Tp*) const [with _Tp = UpdateCheck]’:
/usr/lib/gcc/x86_64-pc-linux-gnu/14/include/g++-v14/bits/unique_ptr.h:399:17:   required from ‘std::unique_ptr<_Tp, _Dp>::~unique_ptr() [with _Tp = UpdateCheck; _Dp = std::default_delete<UpdateCheck>]’
  399 |           get_deleter()(std::move(__ptr));
      |           ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
/mnt/lprup/portage-build/portage/net-im/qtox-9999/work/qtox-9999/src/widget/widget.cpp:151:19:   required from here
  151 |     , nexus{nexus_}
      |                   ^
/usr/lib/gcc/x86_64-pc-linux-gnu/14/include/g++-v14/bits/unique_ptr.h:91:23: error: invalid application of ‘sizeof’ to incomplete type ‘UpdateCheck’
   91 |         static_assert(sizeof(_Tp)>0,
      |                       ^~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

##### Expected Behavior

Build is greater or equal to successful.

--- a/.ci-scripts/build-qtox-linux.sh
+++ b/.ci-scripts/build-qtox-linux.sh
@@ -96,6 +96,7 @@ if [ "$MINIMAL" -eq 1 ]; then
   cmake "$SRCDIR" \
     -DCMAKE_BUILD_TYPE="$BUILD_TYPE" \
     -DSMILEYS=DISABLED \
+    -DUPDATE_CHECK=OFF \
     -DSTRICT_OPTIONS=ON \
     -DSPELL_CHECK=OFF \
     -GNinja \

--- a/src/net/updatecheck.cpp
+++ b/src/net/updatecheck.cpp
@@ -4,7 +4,6 @@
  */
 #include "src/net/updatecheck.h"
 
-#ifdef UPDATE_CHECK_ENABLED
 #include "src/persistence/settings.h"
 #ifndef CMAKE_BUILD
 #include "src/version.h"
@@ -20,6 +19,7 @@
 #include <QTimer>
 #include <cassert>
 
+#ifdef UPDATE_CHECK_ENABLED
 namespace {
 const QString versionUrl{
     QStringLiteral("https://api.github.com/repos/TokTok/qTox/releases/latest")};
@@ -91,8 +91,10 @@ bool isCurrentVersionStable()
 }
 
 } // namespace
+#endif
 
 UpdateCheck::UpdateCheck(const Settings& settings_)
+#ifdef UPDATE_CHECK_ENABLED
     : settings(settings_)
 {
     qInfo() << "qTox is running version:" << GIT_DESCRIBE;
@@ -100,7 +102,13 @@ UpdateCheck::UpdateCheck(const Settings& settings_)
     connect(&updateTimer, &QTimer::timeout, this, &UpdateCheck::checkForUpdate);
     connect(&manager, &QNetworkAccessManager::finished, this, &UpdateCheck::handleResponse);
 }
+#else
+{
+    std::ignore = settings_;
+}
+#endif
 
+#ifdef UPDATE_CHECK_ENABLED
 void UpdateCheck::checkForUpdate()
 {
     if (!settings.getCheckUpdates()) {

--- a/src/net/updatecheck.h
+++ b/src/net/updatecheck.h
@@ -12,7 +12,6 @@
 
 #include <memory>
 
-#ifdef UPDATE_CHECK_ENABLED
 class Settings;
 class QString;
 class QUrl;
@@ -23,6 +22,8 @@ class UpdateCheck : public QObject
 
 public:
     UpdateCheck(const Settings& settings_);
+
+#ifdef UPDATE_CHECK_ENABLED
     void checkForUpdate();
 
 signals:
@@ -38,5 +39,5 @@ private:
     QNetworkAccessManager manager;
     QTimer updateTimer;
     const Settings& settings;
-};
 #endif
+};

--- a/src/widget/form/settings/aboutform.cpp
+++ b/src/widget/form/settings/aboutform.cpp
@@ -52,11 +52,11 @@ AboutForm::AboutForm(UpdateCheck* updateCheck_, Style& style_)
 {
     bodyUI->setupUi(this);
 
-#if !UPDATE_CHECK_ENABLED
+#ifndef UPDATE_CHECK_ENABLED
     bodyUI->updateStack->setVisible(false);
 #endif
     bodyUI->unstableVersion->setVisible(false);
-#if UPDATE_CHECK_ENABLED
+#ifdef UPDATE_CHECK_ENABLED
     connect(updateCheck_, &UpdateCheck::versionIsUnstable, this, &AboutForm::onUnstableVersion);
 #endif
 
@@ -89,7 +89,7 @@ void AboutForm::replaceVersions()
 
     bodyUI->youAreUsing->setText(tr("You are using qTox version %1.").arg(QString(GIT_DESCRIBE)));
 
-#if UPDATE_CHECK_ENABLED
+#ifdef UPDATE_CHECK_ENABLED
     if (updateCheck != nullptr) {
         connect(updateCheck, &UpdateCheck::updateAvailable, this, &AboutForm::onUpdateAvailable);
         connect(updateCheck, &UpdateCheck::upToDate, this, &AboutForm::onUpToDate);

--- a/src/widget/form/settingswidget.cpp
+++ b/src/widget/form/settingswidget.cpp
@@ -57,7 +57,7 @@ SettingsWidget::SettingsWidget(UpdateCheck* updateCheck, IAudioControl& audio, C
     std::unique_ptr<AdvancedForm> expfrm(new AdvancedForm(settings, style, messageBoxManager));
     std::unique_ptr<AboutForm> abtfrm(new AboutForm(updateCheck, style));
 
-#if UPDATE_CHECK_ENABLED
+#ifdef UPDATE_CHECK_ENABLED
     if (updateCheck != nullptr) {
         connect(updateCheck, &UpdateCheck::updateAvailable, this, &SettingsWidget::onUpdateAvailable);
     } else {

--- a/src/widget/widget.cpp
+++ b/src/widget/widget.cpp
@@ -287,13 +287,13 @@ void Widget::init()
     addFriendForm = new AddFriendForm(core->getSelfId(), settings, style, *messageBoxManager, *core);
     conferenceInviteForm = new ConferenceInviteForm(settings, *core);
 
-#if UPDATE_CHECK_ENABLED
+#ifdef UPDATE_CHECK_ENABLED
     updateCheck = std::unique_ptr<UpdateCheck>(new UpdateCheck(settings));
     connect(updateCheck.get(), &UpdateCheck::updateAvailable, this, &Widget::onUpdateAvailable);
 #endif
     settingsWidget = new SettingsWidget(updateCheck.get(), audio, core, *smileyPack, cameraSource,
                                         settings, style, *messageBoxManager, profile, this);
-#if UPDATE_CHECK_ENABLED
+#ifdef UPDATE_CHECK_ENABLED
     updateCheck->checkForUpdate();
 #endif
 

